### PR TITLE
Improve Kubernetes access test coverage

### PR DESCRIPTION
**Pull request by tigrato**
This PR improves the Kubernetes Access test coverage from 61% to >73%.

There are some other code paths that require special attention such as watching with filtering but those require more work.

### CODE REVIEW

File: lib/kube/proxy/kube_creds_test.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2022 Gravitational, Inc.
```
Comment (r0mant) [inline lib/kube/proxy/kube_creds_test.go:2]:
```suggestion
Copyright 2023 Gravitational, Inc.
```

File: lib/kube/proxy/kube_creds_test.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+	"k8s.io/client-go/rest"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/cloud"
+	"github.com/gravitational/teleport/lib/cloud/azure"
+	"github.com/gravitational/teleport/lib/cloud/gcp"
+	"github.com/gravitational/teleport/lib/fixtures"
+)
+
+func Test_DynamicKubeCreds(t *testing.T) {
```
Comment (r0mant) [inline lib/kube/proxy/kube_creds_test.go:49]: Nit: Can you add godoc explaining what the test verifies?

File: lib/kube/proxy/kube_creds_test.go [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+			Endpoint: aws.String("https://api.eks.us-west-2.amazonaws.com"),
+			Name:     req.Name,
+			CertificateAuthority: &eks.Certificate{
+				Data: aws.String(base64.RawStdEncoding.EncodeToString([]byte(fixtures.TLSCACertPEM))),
+			},
+		},
+	}, nil
+}
+
+type stsMockClient struct {
```
Comment (r0mant) [inline lib/kube/proxy/kube_creds_test.go:223]: We have lib/cloud/mocks/aws.go. Can we use STS mock we already have there, and move EKS mock there as well?

File: lib/kube/proxy/kube_creds_test.go [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+		HTTPRequest: &http.Request{
+			Header: http.Header{},
+			URL:    s.u,
+		},
+		Operation: &request.Operation{},
+		Handlers:  request.Handlers{},
+	}, nil
+}
+
+type gkeMockCLient struct {
```
Comment (r0mant) [inline lib/kube/proxy/kube_creds_test.go:239]: Same for GCP/Azure mocks. Let's move them to lib/cloud/mocks for better reuse?

**Review by r0mant — APPROVED** [review]

**Review by mdwn — APPROVED** [review]

**Pull request merged**